src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -87,4 +108,8 @@ def self.initialize_puppet(puppet_config)
   def self.oid_defns
     @@oid_defns
   end
+
+  def self.master_run_mode
+    proc { Puppet::Util::RunMode[:master] }
+  end
```

Comment:
Why did this need to become a proc?

Reply:
It's called in both the `initialize_puppet_settings` and  `initialize_puppet`, and I didn't want to store any state.

Reply:
I don't think I follow.

I also don't think it's a valid thing to change/have changed during a running server.

Reply:
Looking at the implementation of [Puppet::Util::RunMode](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/run_mode.rb), it doesn't look like calling it would ever change the value; a UnixRunMode object just gets stored in the Runmode class in the `@run_modes` hash.

I guess you could say this is slightly over-refactored; I wasn't quite sure what kind of object was coming back, and so I stored it in that proc since I needed it in two different spots. I don't think it needs to be a proc...but I don't think it hurts either?